pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -729,3 +730,47 @@ func generateServerURL(serverIP string, nodePort int32) (string, error) {
 func SupportedStorageMode() []string {
 	return []string{etcdStorageModeEmptyDir, etcdStorageModeHostPath, etcdStorageModePVC}
 }
+
+// NewDefaultCommandInitOption returns a CommandInitOption with default values
+func NewDefaultCommandInitOption() *CommandInitOption {
+	return &CommandInitOption{
+		ImagePullPolicy:                    string(corev1.PullIfNotPresent),
+		KubeImageTag:                       "v1.29.6",
+		Namespace:                          "karmada-system",
+		EtcdStorageMode:                    "hostPath",
+		EtcdInitImage:                      DefaultInitImage,
+		EtcdReplicas:                       1,
+		EtcdHostDataPath:                   "/var/lib/karmada-etcd",
+		EtcdPersistentVolumeSize:           "5Gi",
+		CRDs:                               DefaultCrdURL,
+		KarmadaAPIServerNodePort:           32443,
+		KarmadaDataPath:                    "/etc/karmada",
+		KarmadaPkiPath:                     "/etc/karmada/pki",
+		KarmadaSchedulerImage:              DefaultKarmadaSchedulerImage,
+		KarmadaSchedulerReplicas:           1,
+		KarmadaControllerManagerImage:      DefaultKarmadaControllerManagerImage,
+		KarmadaControllerManagerReplicas:   1,
+		KarmadaWebhookImage:                DefaultKarmadaWebhookImage,
+		KarmadaWebhookReplicas:             1,
+		KarmadaAggregatedAPIServerImage:    DefaultKarmadaAggregatedAPIServerImage,
+		KarmadaAggregatedAPIServerReplicas: 1,
+		WaitComponentReadyTimeout:          cmdinitoptions.WaitComponentReadyTimeout,
+		CertValidity:                       cert.Duration365d,
+		HostClusterDomain:                  globaloptions.DefaultHostClusterDomain,
+	}
```

Comment:
func `NewDefaultCommandInitOption` is intended for use with the `config` command, but the `config` command actually only uses a few of the options. Is it necessary to set default values for all options?

Reply:
I think it could be used as a general initialization operation to facilitate the addition of other commands related to karmadactl config in the future. Adding all the default information here might be helpful for future development work, but removing this information and adding it only when needed is also an option. Which approach do you think is better?

Reply:
> Adding all the default information here might be helpful for future development work, but removing this information and adding it only when needed is also an option. Which approach do you think is better?

I perfer the latter. If subsequent development requires a complete default initialization operation, we can extract the operations from `flags.XXXVarP` to obtain a generic initialization operation that includes default values.

Reply:
> > Adding all the default information here might be helpful for future development work, but removing this information and adding it only when needed is also an option. Which approach do you think is better?
> 
> I perfer the latter. If subsequent development requires a complete default initialization operation, we can extract the operations from `flags.XXXVarP` to obtain a generic initialization operation that includes default values.

I have fixed the defects mentioned above, how does the code look now